spec/policy_machine_storage_adapters/active_record_spec.rb

```diff
@@ -26,21 +26,65 @@
     let(:policy_machine_storage_adapter) { described_class.new }
 
     describe 'find_all_of_type' do
+      it 'accepts an array parameter on a column attribute' do
+        search_uuids = ['some_uuid1', 'some_uuid2']
+        policy_machine_storage_adapter.add_object(search_uuids[0], 'some_policy_machine_uuid1')
+        policy_machine_storage_adapter.add_object(search_uuids[1], 'some_policy_machine_uuid1')
+        policy_machine_storage_adapter.add_object('some_uuid3', 'some_policy_machine_uuid1')
+
+        expect(
+          policy_machine_storage_adapter.find_all_of_type_object(
+            unique_identifier: search_uuids
+          ).count
+        ).to eq(2)
+      end
+
+      context 'when case insensitive' do
+        it 'accepts an array parameter on a column attribute' do
+          colors = ['some_uuid1', 'some_uuid2']
+          policy_machine_storage_adapter.add_object('some_uuid1', 'some_policy_machine_uuid1', color: colors[0])
+          policy_machine_storage_adapter.add_object('some_uuid2', 'some_policy_machine_uuid1', color: colors[1])
+          policy_machine_storage_adapter.add_object('some_uuid3', 'some_policy_machine_uuid1', color: nil)
 
-      it 'warns once when filtering on an extra attribute' do
-        expect(Warn).to receive(:warn).once
-        2.times do
-          expect(policy_machine_storage_adapter.find_all_of_type_user(foo: 'bar')).to be_empty
+          expect(
+            policy_machine_storage_adapter.find_all_of_type_object(
+              color: colors,
+              ignore_case: true
+            ).count
+          ).to eq(2)
         end
       end
 
-      context 'an extra attribute column has been added to the database' do
+      it 'allows uuid as a parameter' do
+        uuid = 'some_uuid'
+        policy_machine_storage_adapter.add_object(uuid, 'some_policy_machine_uuid')
 
+        expect(
+          policy_machine_storage_adapter.find_all_of_type_object(uuid: uuid).count
+        ).to eq(1)
+      end
+
+      context 'an extra attribute column has been added to the database' do
```

Comment:
This is a little misleading, `extra_attributes` are faux column attributes by definition.